### DISCUSSION TITLE
Benchmarking on intel compilers?

### DISCUSSION THREAD

**Discussion by gdevenyi** [zlib-ng/zlib-ng#856] [General]
I have access to a number of intel compilers on my HPC as well as the ability to grab a dedicated machine to avoid CPU contention.

Would anyone be interested in some benchmarking?

If so, please point me at some scripts and I'll be happy to run and post the results.

**Comment by neurolabusc** [comment]
@gdevenyi - I created this simple Python [benchmark for you](https://github.com/neurolabusc/zlib-bench-python). So far, it tests gcc and clang (I do not have the Intel compiler installed). You should be able to edit the script for your needs. It test the popular [silesia](http://sun.aei.polsl.pl/~sdeor/index.php?page=silesia) corpus.

**Comment by gdevenyi** [comment]
Thanks @neurolabusc,

here's the output:
![image](https://user-images.githubusercontent.com/3001850/83341788-b18fe800-a2b5-11ea-847f-721ad0374b91.png)

**Comment by neurolabusc** [comment]
Nice. I have updated the script to optionally also test CloudFlare. It is curious that for zlib-ng gcc seems to perform worse than Clang, ICC and [MSVC](https://github.com/zlib-ng/zlib-ng/pull/588). However, for CloudFlare gcc outperforms Clang. I wonder if your comment regarding [-O2 vs -O3](https://github.com/zlib-ng/zlib-ng/issues/515) might explain this. Perhaps at -O3 gcc can catch up to the other compilers. Theres not a huge difference, but testing different compilers is good for regression testing.

I still find it remarkable that for this corpus zlib-ng compresses more poorly for compression levels 7-9 versus 6. I realize there are different strategies, but the classic zlib does seem to provide a more reliable speed-compression trade off.

![silesia_speed_size](https://user-images.githubusercontent.com/8930807/83352419-3102d300-a319-11ea-9ca3-78879bf273da.png)

**Comment by gdevenyi** [comment]
Thanks for the update @neurolabusc. Here's an updated plot.
![image](https://user-images.githubusercontent.com/3001850/83364486-94b7eb00-a36f-11ea-88dc-27b930a6bfb6.png)
